src/main/java/io/kestra/plugin/gcp/firestore/Query.java

```diff
@@ -0,0 +1,230 @@
+import io.kestra.core.models.annotations.Example;
+import io.kestra.core.models.annotations.Plugin;
+import io.kestra.core.models.annotations.PluginProperty;
+import io.kestra.core.models.executions.metrics.Counter;
+import io.kestra.core.models.tasks.RunnableTask;
+import io.kestra.core.runners.RunContext;
+import io.kestra.core.serializers.FileSerde;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+import org.apache.commons.lang3.tuple.Pair;
+
+import javax.validation.constraints.NotNull;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static io.kestra.core.utils.Rethrow.throwConsumer;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+        title = "Query documents of a collection.",
+        description = "Query documents of a collection."
+)
+@Plugin(
+        examples = {
+                @Example(
+                        title = "Query documents of a collection.",
+                        code = {
+                                "collection: \"persons\"",
+                                "field: \"firstname\"",
+                                "value: \"Doe\""
+                        }
+                )
+        }
+)
+public class Query extends AbstractFirestore implements RunnableTask<Query.Output> {
+
+    @Schema(
+            title = "Whether to store the data from the query result into an ion serialized data file."
+    )
+    @PluginProperty(dynamic = false)
+    @Builder.Default
+    private Boolean store = false;
+
+    @Schema(
+            title = "Field name for the where clause."
+    )
+    @PluginProperty(dynamic = false)
+    @NotNull
+    private String field;
+
+    @Schema(
+            title = "Field value for the where clause.",
+            description = "Field value for the where clause. Only strings are supported at the moment."
+    )
+    @PluginProperty(dynamic = true)
+    @NotNull
+    private String value;
+
+    @Schema(
+            title = "The query operator for the where clause, by default EQUAL_TO that will call 'collection.whereEqualTo(name, value)'",
+            description = "Can be one of EQUAL_TO, " +
+                    "NOT_EQUAL_TO, " +
+                    "LESS_THAN, " +
+                    "LESS_THAN_OR_EQUAL_TO, " +
+                    "GREATER_THAN, " +
+                    "GREATER_THAN_OR_EQUAL_TO"
+    )
+    @PluginProperty(dynamic = false)
+    @Builder.Default
+    private QueryOperator queryOperator = QueryOperator.EQUAL_TO;
+
+    @Schema(
+            title = "Field name for the order by clause."
+    )
+    @PluginProperty(dynamic = false)
+    private String orderBy;
+
+    @Schema(
+            title = "Start offset for pagination of the query results."
+    )
+    @PluginProperty(dynamic = true)
+    private Integer offset;
```

Comment:
@tchiotludo When a dynamic parameter is an Integer we cannot render it with the runContext, will it works this way ? The MongoDB Plugin does the same.

Reply:
You fell on the dark side of the rendering, we have an issue with int, double, etc ... that can't used `runContext.render` and this can be annoying when using variable. Maybe a ticket on that side to deal it on the future and for now `@PluginProperty(dynamic = false)`

Reply:
Thinking about it I think we don't have another solution than
- Inject it as String
- Use the runContext to render it
- Parse it
I'll open an issue about this, and we may need to check on all existing plugins that declare a dynamic property and inject it in a primitive wrapper.